server/src/main/java/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -1324,6 +1324,11 @@ public void cleanupStorage(boolean recurring) {
                             s_logger.warn("Unable to destroy uploaded template " + template.getUuid() + ". Error details: " + th.getMessage());
                         }
                     }
+                    List<VMTemplateVO> vmTemplateVOS = _templateDao.listUnRemovedTemplatesByStates(VirtualMachineTemplate.State.Inactive);
+                    for (VMTemplateVO template: vmTemplateVOS) {
+                        template.setRemoved(new Date());
+                        _templateDao.update(template.getId(), template);
+                    }
```

Comment:
Can you extract this? method already reaches from l1156-1339. ( a `class StorageCleaner` required :| )

Reply:
done @DaanHoogland